pkg/network/common/egressip.go

```diff
@@ -290,6 +300,174 @@ func (eit *EgressIPTracker) UpdateHostSubnetEgress(hs *networkv1.HostSubnet) {
 	eit.syncEgressIPs()
 }
 
+func nodeIsReady(node *v1.Node) bool {
+	for _, cond := range node.Status.Conditions {
+		if cond.Type == v1.NodeReady {
+			if cond.Status == v1.ConditionFalse || cond.Status == v1.ConditionUnknown {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+// SetMonitorNodes updates EgressIPTracker with the node data provided and
+// starts a go-routine (if one is not already running) which polls monitorNodes
+// to check if they are ready and reachable.
+func (eit *EgressIPTracker) SetMonitorNodes(monitorNodes []*NodeEgress) {
+	for _, node := range monitorNodes {
+		if _, exists := eit.monitorNodes.Load(node.NodeIP); !exists {
+			eit.monitorNodes.Store(node.NodeIP, node)
+		}
+	}
+	eit.monitorNodes.Range(func(key, value interface{}) bool {
+		nodeIP := key.(string)
+		found := false
+		for _, node := range monitorNodes {
+			if node.NodeIP == nodeIP {
+				found = true
+			}
+		}
+		if !found {
+			eit.monitorNodes.Delete(nodeIP)
+		}
+		return true
+	})
+	if eit.getMonitorNodesLen() > 0 {
+		if eit.stop == nil {
+			eit.stop = make(chan struct{})
+			go utilwait.PollUntil(defaultPollInterval, eit.poll, eit.stop)
+		}
+	} else {
+		if eit.stop != nil {
+			close(eit.stop)
+			eit.stop = nil
+		}
+	}
+}
+
+// AddEgressIP starts a go-routine (if one is not already running) which polls
+// monitorNodes to check if they are ready and reachable
+func (eit *EgressIPTracker) AddEgressIP(nodeIP, nodeName, egressIP string) {
+	if nodeIntf, exists := eit.monitorNodes.Load(nodeIP); exists {
+		node := nodeIntf.(*NodeEgress)
+		node.activeEgressIPs.Insert(egressIP)
+		return
+	}
+	klog.V(4).Infof("Monitoring node %s", nodeIP)
+	eit.monitorNodes.Store(nodeIP, &NodeEgress{
+		NodeIP:          nodeIP,
+		activeEgressIPs: sets.NewString(egressIP),
+		NodeName:        nodeName,
+	})
+	if eit.getMonitorNodesLen() == 1 && eit.stop == nil {
+		eit.stop = make(chan struct{})
+		go utilwait.PollUntil(defaultPollInterval, eit.poll, eit.stop)
+	}
+}
+
+func (eit *EgressIPTracker) getMonitorNodesLen() int {
```

Comment:
The fact that you have to implement this by hand (in this way) suggests that a `sync.Map` isn't really the data structure we want here. Why did you change this from a mutex+map again? I don't see that in the comments.